Couchers-org/couchers
# New Events Page - Functionality

This PR adds all the functionality for the events page.

**NOTE:** I did not add the "Friends" filters under `Discover` as there is no backend functionality for it yet.

There will be one final PR where I add the tests and the storybook components.

**TO TEST:**
- You need to make a variety of events - canceled, online, make 1-2 in the past by setting to 1 min from now (and waiting 1 min lol), different locations. If possible log in to a second account and make events from another user to test the "my events" part (user that's not you should only show under discover if you're not attending).
- Click around the filters, upcoming, past, use the search input, etc.
- Test pagination, i.e. if you are on last page and change filter, it should go back to first page.
- Make sure "My Events" on the dashboard page not changed

**Web frontend checklist**
- [ x ] Formatted my code with `yarn format`
- [ x ] There are no warnings from `yarn lint --fix`
- [ x ] There are no console warnings when running the app
- [ NEXT PR ] Added any new components to storybook
- [ NEXT PR ] Added tests where relevant
- [ x ] All tests pass
- [ x ] Clicked around my changes running locally and it works
- [ x ] Checked Desktop, Mobile and Tablet screen sizes
